viewer/src/utilities/networking/CogniteClient3dExtensions.ts

```diff
@@ -2,105 +2,67 @@
  * Copyright 2020 Cognite AS
  */
 
-import { CogniteClient, CogniteInternalId, CogniteExternalId, IdEither } from '@cognite/sdk';
+import { CogniteClient, IdEither, ItemsResponse } from '@cognite/sdk';
+import { HttpHeaders } from '@cognite/sdk/dist/src/utils/http/basicHttpClient';
 import { File3dFormat } from '../File3dFormat';
-
-export type Model3dOutput = {
-  readonly format: File3dFormat | string;
-  readonly version: number;
-  readonly blobId: CogniteInternalId;
-};
-
-export class Model3dOutputList {
-  public readonly model: CogniteModel3dIdentifier;
-  public readonly outputs: Model3dOutput[];
-
-  constructor(model: CogniteModel3dIdentifier, outputs: Model3dOutput[]) {
-    this.model = model;
-    this.outputs = outputs;
-  }
-
-  /**
-   * Finds an output with a given format of the most recent version.
-   *
-   * @param outputFormat        Format to find output for, either a well known format a custom format.
-   * @param supportedVersions   Optional list of supported version. If not provided all versions are considered.
-   */
-  public findMostRecentOutput(
-    outputFormat: File3dFormat | string,
-    supportedVersions?: number[]
-  ): Model3dOutput | undefined {
-    const candidates = this.outputs.filter(
-      x => x.format === outputFormat && (!supportedVersions || supportedVersions.indexOf(x.version) !== -1)
-    );
-    return candidates.length > 0
-      ? candidates.reduce((left, right) => {
-          return right.version > left.version ? right : left;
-        })
-      : undefined;
-  }
-}
-
-type CogniteModel3dIdentifier = { id: CogniteInternalId } | { externalId: CogniteExternalId };
+import { CadSceneProvider } from '@/dataModels/cad/internal/CadSceneProvider';
+import { ModelUrlProvider } from './ModelUrlProvider';
+import { BlobOutputMetadata } from './BlobOutputMetadata';
+import { ModelOutputList } from './ModelOutputList';
+import { CadSectorProvider } from '@/dataModels/cad/internal/sector/CadSectorProvider';
 
 interface OutputsRequest {
   models: IdEither[];
   formats?: (string | File3dFormat)[];
 }
 
-interface OutputsResponse {
-  readonly items: {
-    readonly model: CogniteModel3dIdentifier;
-    readonly outputs: Model3dOutput[];
-  }[];
-}
-
 /**
  * Provides 3D V2 specific extensions for the standard CogniteClient used by Reveal.
  */
-export class CogniteClient3dExtensions {
+export class CogniteClient3dExtensions implements ModelUrlProvider, CadSceneProvider, CadSectorProvider {
```

Comment:
I think what you maybe are doing here is to create a specialized version for Reveal which makes sense, but I think `CogniteClient3dExtensions` should be a class that we can migrate to `@cognite/sdk` and hence closely map to the API. It's probably a good idea with something that wraps this to offer the functionality implemented here.